### PR TITLE
Add Radix to IntrinsicScalarFunction

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Towards: https://github.com/lfortran/lfortran/issues/2208

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
@meow464 I think we have it.
Do you want to make any changes?

**Comment by certik** [comment]
Register the test in cmake.

**Comment by meow464** [comment]
> @meow464 I think we have it.
> 
> Do you want to make any changes here?

I think it looks great. Thanks for the help!

**Review by czgdp1807 — APPROVED** [review]
LGTM.

File: src/libasr/pass/intrinsic_function_registry.h [hunk 1]
```diff
@@ -1260,6 +1261,33 @@ namespace Abs {
...
+    // Function to create an instance of the 'radix' intrinsic function
+    static inline ASR::asr_t* create_Radix(Allocator& al, const Location& loc,
+        Vec<ASR::expr_t*>& args,
+        const std::function<void (const std::string &, const Location &)> err) {
+        if ( args.n != 1 ) {
+            err("Intrinsic `radix` accepts exactly one argument", loc);
+        } else if ( !is_real(*expr_type(args[0]))
+                 && !is_integer(*expr_type(args[0])) ) {
+            err("Argument of the `radix` must be Integer or Real", loc);
+        }
```
Comment (czgdp1807) [inline src/libasr/pass/intrinsic_function_registry.h:1282]: Let's add an error test for this.

**Review by czgdp1807 — DISMISSED** [review]

File: src/libasr/pass/intrinsic_function_registry.h (hunk 1 above)
Comment (czgdp1807) [reply src/libasr/pass/intrinsic_function_registry.h:1282]: Pass complex number and expect an error.

**Pull request merged**